indico/htdocs/js/indico/Core/Dialogs/Users.js

```diff
@@ -1481,18 +1481,37 @@ type("UserListField", ["IWidget"], {
 
     draw: function() {
         var self = this;
-        var select;
-        var buttonDiv = Html.div({style:{marginTop: pixels(10)}});
+        var buttonDiv = Html.div({style: {marginTop: pixels(10)}});
 
         if (this.allowSearch || this.includeFavourites || exists(this.suggestedUsers)) {
+            var chooseUserButton = Html.input("button", {style: {marginRight: pixels(5)},
+                                                         className: 'i-button', type: 'button'},
+                                              this.enableGroups ? $T('Add Indico User / Group'): $T('Add Indico user'));
+            var chooseIpNetwork = Html.a({href: '#', style: {marginRight: pixels(5)},
+                                          className: 'i-button arrow js-dropdown',
+                                          'data-toggle': 'dropdown'}, $T('Add IP Network'));
+            var ipNetworksContainer = Html.span({className: 'group'});
+            var ipNetworksList = Html.ul({className: 'dropdown'});
+
+            each(this.ipNetworks, function(network) {
+                var li = Html.li();
+                li.append(Html.a({href: '#'}, network.name));
+                li.observeClick(function() {
+                    self.newProcess([network], function(result) {
+                        if (result && !self.userList.get(network.identifier)) {
+                            self.userList.set(network.identifier, $O(network))
+                        }
+                    });
+                });
+                ipNetworksList.append(li);
+            });
 
-            var chooseUserButton = Html.input("button", {style:{marginRight: pixels(5)}, className: 'i-button'}, this.enableGroups ? $T('Add Indico User / Group'): $T('Add Indico user'));
-
-            var title = "";
+            ipNetworksContainer.append(chooseIpNetwork);
+            ipNetworksContainer.append(ipNetworksList);
             if (this.includeFavourites || exists(this.suggestedUsers)) {
-                title = this.enableGroups ? $T("Add Users and Groups") : $T("Add Users");
+                var title = this.enableGroups ? $T("Add Users and Groups") : $T("Add Users");
```

Comment:
the `var title;` should stay outside. repeating the same `var foo` twice in the same scope is not very clean since it's hoisted to the top anyway